FormatData.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+abstract class Format
+{
+  public $data;
+  function __construct($data)
+  {
+    $this->data = $data;
+  }
+  abstract public function formatData();
+}
+
+class Json extends Format
+{
+  public function formatData()
+  {
+    $json_data = json_encode($this->data);
```

Comment:
if you don't use a variable you added so no need to reverse new space in the memory, please return early